app/full_text_search.py

```diff
@@ -0,0 +1,87 @@
+import logging
+import re
+
+from google.appengine.api import search as appengine_search
+
+import model
+
+# The index name for full text search
+PERSON_FULLTEXT_INDEX_NAME = 'person_information'
+
+def search(repo, query_obj, max_results):
+    """
+    Searches person with index.
+    Args:
+        repo: The name of repository
+        query_obj: Search word
+        max_results: The max results you want.(Maximum: 1000)
+
+    Returns:
+        results[<model.Person>, ...]
+
+    Raises:
+        search.Error: An error occurred search unknown index, syntax error
+    """
+    #TODO: Sanitaize query_obj
+    results = []
+    if not query_obj:
+        return results
+    index = appengine_search.Index(name=PERSON_FULLTEXT_INDEX_NAME)
+    options = appengine_search.QueryOptions(
+        limit=max_results,
+        returned_fields=['record_id'])
+    index_results = index.search(appengine_search.Query(
+        query_string=query_obj, options=options))
+    record_ids = []
+    for document in index_results:
+        record_ids.append(document.fields[0].value)
+    for id in record_ids:
+        results.append(model.Person.get_by_key_name(repo + ':' + id))
+    return results
+
+
+def create_document(**kwargs):
+    """
+    Creates document for full text search.
+    It should be called in add_record_to_index method.
+    """
+    doc_id = kwargs['repo'] + ':' + kwargs['record_id']
+    TEXT_FIELD_TABLE = ['record_id', 'repo', 'given_name', 'family_name',
```

Comment:
Do you need a list of fields here? Isn't create_document() called only with fields you want to index? One idea is:

* Make it def create_document(doc_id, **kwargs)
* Construct doc_id in add_record_to_index instead of here
* Add all items in kwargs to the fields

But feel free to keep this constant if you think this is useful later when you extend it to add location.